src/pages/EditUser.js

```diff
@@ -0,0 +1,349 @@
+import React, { useRef, useState } from 'react'
+import { Formik} from "formik"
+import * as yup from "yup"
+import { Form, Button, Container, Row, Col, Modal } from "react-bootstrap";
+import { FaUserCircle } from 'react-icons/fa'
+import { IconContext } from 'react-icons/lib'
+import AuthService from "../services/auth.service.js"
+import "../styles/pages/EditUser.css"
+import { useHistory } from 'react-router-dom';
+import { LinkContainer } from 'react-router-bootstrap'
+
+const schema = yup.object().shape({
+
+    email: yup.string().email('Invalid email').nullable(),
+    password: yup.string()
+                .min(4, "Please enter at least 4 characters.")
+                .nullable(),
+    first_name: yup.string()
+                .max(50, 'Too Long!')
+                .nullable(),
+    last_name: yup.string()
+                .max(50, 'Too Long!')
+                .nullable(),
+    contact_number: yup.number()
+                .min(1, "Invalid")
+                .nullable(),
+    gender: yup.string()
+                .nullable(),
+    civil_status: yup.string()
+                    .nullable(),
+    birth_date: yup.string()
+                    .nullable(),
+    home_address: yup.string()
+                .min(4, "Please enter at least 4 characters.")
+                .nullable(),
+                
+})
+
+function EditUser() {
+    console.log(AuthService.getCurrentUser().uuid)
+    const uuid_creds =  AuthService.getCurrentUser().uuid;
+    const formRef = useRef()
+    const history = useHistory();
+    const [show, setShow] = useState(false);
+
+    const handleClose = () => {
+        setShow(false)
+        history.push("/tracevax/qrcodedisplay")
+    };
+    const handleShow = () => setShow(true);
+    const updateUser = () => {
+        
+        AuthService.update(
+            formRef.current.values.email, 
+            formRef.current.values.password, 
+            formRef.current.values.first_name, 
+            formRef.current.values.last_name,
+            formRef.current.values.contact_number, 
+            formRef.current.values.gender, 
+            formRef.current.values.civil_status, 
+            formRef.current.values.birth_date,
+            formRef.current.values.home_address,
+            uuid_creds)
+            .then(response => {
```

Comment:
note that responses are ideally handled inside the service so we can have separation of concerns for services and our UI components.
We should handle this when we refactor the app to use context-providers/redux